simulator does not compile on latest ubuntu

Did GHA just upgrade GCC? No? What's going on? why is build red?

https://github.com/rusefi/rusefi/actions/runs/3493929725/attempts/1
![image](https://user-images.githubusercontent.com/48498823/202613841-218a1f11-29ee-4c3f-9ff0-43a7e3f215ea.png)


previous build

![image](https://user-images.githubusercontent.com/48498823/202613634-1a5165a9-b455-4133-a541-a3bb92fbed58.png)

![image](https://user-images.githubusercontent.com/48498823/202613697-83244136-646c-47a7-b695-06b70512cbc5.png)

FYI, `master` (717a51095c67964b99fdc828c650c7e463cd162b) fails to build w/ GCC 12:
```
Compiling hal_can_lld.cpp
simulator/can/hal_can_lld.cpp: In function ‘void can_lld_stop(CANDriver*)’:
simulator/can/hal_can_lld.cpp:143:14: error: ‘remove’ is not a member of ‘std’; did you mean ‘move’?
  143 |         std::remove(instances.begin(), instances.end(), canp);
      |              ^~~~~~
      |              move
make: *** [../firmware/ChibiOS/os/common/startup/SIMIA32/compilers/GCC/rules.mk:130: build/obj/hal_can_lld.o] Error 1
Simulator compilation failed
```

--- a/.github/workflows/build-firmware.yaml
+++ b/.github/workflows/build-firmware.yaml
@@ -4,7 +4,7 @@ on: [push, pull_request, workflow_dispatch]
 
 jobs:
   build-firmware:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
 
     strategy:
       # Let all builds finish even if one fails early
@@ -406,7 +406,7 @@ jobs:
 
   # here we build f407-discovery
   build-primary-bundle:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     steps:
     - uses: actions/checkout@v3
       with:

